### PR TITLE
resolves #507 publish contact after identity creation wizard

### LINKED ISSUE #507: Publish contact data of identity after using the wizard..
..or make clear that this won't happen automatically.

I prefer to publish the contact data (if and only if an mail address OR a phone number has been given). This would be the same behavior as in Android. And it would comply to the texts mentioned in bug #506 (otherwise the texts mentioned there are wrong). #506 Should be fixed before this one.

### PR DESCRIPTION
**Pull request by julianseeger**
When you look at the test, you see the implementation of #507

While implementing that, I realized that the labels are too small for the new, long instructions and I had to restyle the wizard (a little bit more flexible...)

### CODE REVIEW

File: src/main/java/de/qabel/desktop/ui/accounting/wizard/WizardController.java [hunk 1]
```diff
@@ -123,20 +156,42 @@ private void addListeners() {
...
+            () -> validatePhone(phoneInput.getText()),
+            phoneInput.textProperty()
+        );
+        nextButton.disableProperty().bind(
+            hasAlias.not().or(
+                currentStepIndex.isEqualTo(1).and(hasEmail).and(emailValid.not())
+            ).or(
+                currentStepIndex.isEqualTo(2).and(hasPhone).and(phoneValid.not())
+            )
+        );
```
Comment (julianseeger) [inline src/main/java/de/qabel/desktop/ui/accounting/wizard/WizardController.java]: Disable next button if:
 - on second (1) step and there is an invalid email or
 - on third (2) step and there is an invalid phone number

File: src/main/java/de/qabel/desktop/ui/accounting/wizard/WizardController.java [hunk 2]
```diff
@@ -153,6 +208,7 @@ private void createSteps() {
         steps.addAll(Arrays.asList(
             step1, step2, step3, step4
         ));
+        steps.forEach(step -> step.managedProperty().bind(step.visibleProperty()));
```
Comment (julianseeger) [inline src/main/java/de/qabel/desktop/ui/accounting/wizard/WizardController.java]: remove invisible steps from layout

File: src/main/java/de/qabel/desktop/ui/accounting/wizard/WizardController.java [hunk 3]
```diff
@@ -194,17 +250,28 @@ private void loadCurrentStep() {
...
+                child.requestFocus();
+                return;
+            }
+        }
     }
 
     protected Boolean validatePhone(String phone) {
         try {
             String formattedPhone = PhoneUtilsKt.formatPhoneNumber(phone);
+            phoneLabel.setText(formattedPhone);
```
Comment (julianseeger) [inline src/main/java/de/qabel/desktop/ui/accounting/wizard/WizardController.java]: show invalid phone numbers (let users detect their errors / typos / wrong locale configuration)

File: src/main/java/de/qabel/desktop/ui/accounting/wizard/WizardController.java [hunk 4]
```diff
@@ -240,8 +307,14 @@ public void buttonClosePopup() {
 
     public void finishWizard() {
         if (identity != null && !aliasInput.getText().isEmpty()) {
+            boolean uploadEnabled = !identity.getEmail().isEmpty() || !identity.getPhone().isEmpty();
+            identity.setUploadEnabled(uploadEnabled);
             saveIdentity();
             clientConfiguration.selectIdentity(identity);
+
+            if (uploadEnabled) {
+                new Thread(() -> indexService.updateIdentity(identity, null)).start();
```
Comment (audax) [inline src/main/java/de/qabel/desktop/ui/accounting/wizard/WizardController.java]: Where are potentials errors from the upload shown?

**Pull request merged**